src/hotspot/share/classfile/vmSymbols.cpp

```diff
@@ -296,3 +298,43 @@ vmSymbolID vmSymbols::find_sid(const char* symbol_name) {
   if (symbol == nullptr)  return vmSymbolID::NO_SID;
   return find_sid(symbol);
 }
+
+// The list of these migrated value classes is in
+// open/make/modules/java.base/gensrc/GensrcValueClasses.gmk.
+
+Symbol* vmSymbols::_migrated_class_names[_migrated_class_names_length];
+
+void vmSymbols::initialize_migrated_class_names() {
+  int i = 0;
+  _migrated_class_names[i++] = java_lang_Byte();
+  _migrated_class_names[i++] = java_lang_Short();
+  _migrated_class_names[i++] = java_lang_Integer();
+  _migrated_class_names[i++] = java_lang_Long();
+  _migrated_class_names[i++] = java_lang_Float();
+  _migrated_class_names[i++] = java_lang_Double();
+  _migrated_class_names[i++] = java_lang_Boolean();
+  _migrated_class_names[i++] = java_lang_Character();
+  _migrated_class_names[i++] = java_lang_Number();
+  _migrated_class_names[i++] = java_lang_Record();
+  _migrated_class_names[i++] = java_util_Optional();
+  _migrated_class_names[i++] = java_util_OptionalInt();
+  _migrated_class_names[i++] = java_util_OptionalLong();
+  _migrated_class_names[i++] = java_util_OptionalDouble();
+  _migrated_class_names[i++] = java_time_LocalDate();
+  _migrated_class_names[i++] = java_time_LocalDateTime();
+  _migrated_class_names[i++] = java_time_LocalTime();
+  _migrated_class_names[i++] = java_time_Duration();
+  _migrated_class_names[i++] = java_time_Instant();
+  _migrated_class_names[i++] = java_time_MonthDay();
+  _migrated_class_names[i++] = java_time_ZonedDateTime();
+  _migrated_class_names[i++] = java_time_OffsetDateTime();
+  _migrated_class_names[i++] = java_time_OffsetTime();
+  _migrated_class_names[i++] = java_time_YearMonth();
+  _migrated_class_names[i++] = java_time_Year();
+  _migrated_class_names[i++] = java_time_Period();
+  _migrated_class_names[i++] = java_time_chrono_ChronoLocalDateImpl();
+  _migrated_class_names[i++] = java_time_chrono_MinguoDate();
+  _migrated_class_names[i++] = java_time_chrono_HijrahDate();
+  _migrated_class_names[i++] = java_time_chrono_JapaneseDate();
+  _migrated_class_names[i++] = java_time_chrono_ThaiBuddhistDate();
```

Comment:
Should this `assert(i == _migrated_class_names_length);` as a handshake with the size of the set of migrated classes declared in vmSymbols.hpp?

Reply:
That's a good idea.  I didn't know a better C++ way to initialize these.  I'm sure it exists.